### PR TITLE
Adding pre-commit with ruff and pytest

### PR DESCRIPTION
**Pull request by ajitkshirsagar**
- Removed flake8 CI check
- Added ruff CI check
- Added pre-commit hooks for `ruff` and `pytest`

Closes https://github.com/MrPowers/quinn/issues/69

### CODE REVIEW

File: .github/workflows/ruff.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - uses: chartboost/ruff-action@v1
+        with:
+          version: 0.0.261
+          line-length: "150"
```
Comment (SemyonSinchenko) [inline .github/workflows/ruff.yml:20]: Do we really need to specify all the Ruff parameters in both workflow and pyproject? For me it is better to have all the configurations in pyproject only!

Comment (ajitkshirsagar) [reply .github/workflows/ruff.yml]: Sorry missed that. Moved the configs to pyproject.

**Comment by MrPowers** [comment]
@SemyonSinchenko - looks like @ajitkshirsagar addressed your comment.  Can we merge this one now?

**Comment by SemyonSinchenko** [comment]
> @SemyonSinchenko - looks like @ajitkshirsagar addressed your comment.  Can we merge this one now?

I was going to make some local experiments with it. I'm just overloaded a little by work now. May it wait until the weekend?

**Comment by MrPowers** [comment]
@SemyonSinchenko - yep, sure thing, thank you!

**Comment by SemyonSinchenko** [comment]
Guys, sorry for the delay. I think we can merge it.

**Comment by MrPowers** [comment]
@ajitkshirsagar - thanks for the contribution!!

**Pull request merged**